src/preferences/common/preference-service.spec.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { JsonPreferenceService, PreferenceService } from './preference-service';
+import { FileSystem, FileStat } from "../../filesystem/common/filesystem";
+import { FileSystemWatcher } from "../../filesystem/common/filesystem-watcher"
+import "mocha";
+import * as chai from "chai";
+import * as chaiAsPromised from "chai-as-promised";
+
+
+const expect = chai.expect;
+let prefService: PreferenceService;
+let fileWatcher: FileSystemWatcher;
+let fs: FileSystem;
+
+before(() => {
+    chai.config.showDiff = true;
+    chai.config.includeStack = true;
+    chai.should();
+    chai.use(chaiAsPromised);
+});
+
+beforeEach(() => {
+    fs = new FileSystemStub();
+    fileWatcher = new FileSystemWatcher();
+    prefService = new JsonPreferenceService(fs, fileWatcher, ".theia/prefs.json");
+});
+
+describe('preference-service', () => {
+
+    describe('01 #has preference', () => {
+        it('should return true for the has preference', () => {
+            return expect(prefService.has("lineNumbers")).to.eventually.equal(true);
+        });
+
+        it('should return false for the has preference', () => {
+            return expect(prefService.has("nonExistingPref")).to.eventually.equal(false);
```

Comment:
> Something is messed up with the config...

For clarification; not necessarily here. Somewhere in the code. Or I am simply too tired to understand it now.